pydicer/preprocess/data.py

```diff
@@ -29,44 +31,55 @@ def preprocess(self):
-            - the hashed patient ID
-            - the hashed study UID
-            - a list of dicts that has 2 keys (path to file, slice location)
-            - the series modality
-            - the series UID that it is linked to
-
-        Ex:
-            {
-                "series_uid": {
-                    "patient_id": "",
-                    "study_id": "",
-                    "files": [],
-                    "modality": "",
-                    "linked_series_uid": {}
-                }
-            }
+        Returns: res_dict (pd.DataFrame): containing a row for each DICOM file that was
+           preprocessed, with the following columns:
+            - patient_id: PatientID field from the DICOM header
+            - study_uid: StudyInstanceUID field from the DICOM header
+            - series_uid: SeriesInstanceUID field from the DICOM header
+            - modality: Modailty field from the DICOM header
+            - sop_class_uid: SOPClassUID field from the DICOM header
+            - for_uid: FrameOfReferenceUID field from the DICOM header
+            - file_path: The path to the file (as a pathlib.Path object)
+            - slice_location: The real-world location of the slice (used for imaging modalities)
+            - referenced_series_uid: The SeriesUID referenced by this DICOM file (used for RTSTRUCT
+              and RTDOSE)
+            - referenced_for_uid: The ReferencedFrameOfReferenceUID referenced by this DICOM file
+
         """
-        res_dict = {}
+        df = pd.DataFrame(
+            columns=[
+                "patient_id",
+                "study_uid",
+                "series_uid",
+                "modality",
+                "sop_class_uid",
+                "for_uid",
+                "file_path",
+                "slice_location",
+                "referenced_series_uid",
+                "referenced_for_uid",
+            ]
+        )
         files = self.working_directory.glob("**/*.dcm")
 
         for file in files:
             ds = pydicom.read_file(file, force=True)
 
-            linked_series_uid = {}
-
             try:
 
                 dicom_type_uid = ds.SOPClassUID
 
-                if ds.SeriesInstanceUID not in res_dict:
-                    res_dict[ds.SeriesInstanceUID] = {
-                        "patient_id": ds.PatientID,
-                        "study_id": ds.StudyInstanceUID,
-                        "files": [],
-                        "modality": ds.Modality,
-                        "sop_class_uid": dicom_type_uid,
-                    }
+                res_dict = {
+                    "patient_id": ds.PatientID,
+                    "study_uid": ds.StudyInstanceUID,
+                    "series_uid": ds.SeriesInstanceUID,
+                    "modality": ds.Modality,
+                    "sop_class_uid": dicom_type_uid,
+                    "file_path": str(file),
```

Comment:
I was going to suggest that we make the exported file path of type String, as I was facing issues trying to export into JSON with the PathPosix object, nice!